File: leetcode/cpp/2932_Maximum_Strong_Pair_XOR_I.py
from typing import List

class Solution:
    def maximumStrongPairXor(self, nums: List[int]) -> int:
        n = len(nums)
        mx = 0

        for i in range(n):
            for j in range(i + 1, n):
                x, y = nums[i], nums[j]

                if abs(x - y) <= min(x, y):
                    mx = max(mx, x ^ y)

        return mx

File: leetcode/cpp/2932_Maximum_Strong_Pair_XOR_I.cpp
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
    int maximumStrongPairXor(vector<int>& nums) {
        int n = nums.size();
        int mx = 0;

        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {

                int x = nums[i], y = nums[j];

                if (abs(x - y) <= min(x, y)) {
                    mx = max(mx, x ^ y);
                }
            }
        }

        return mx;
    }
};
